src/centrality/betweenness.c

```diff
@@ -1018,3 +1015,841 @@ int igraph_edge_betweenness_estimate(const igraph_t *graph, igraph_vector_t *res
+    igraph_integer_t no_of_nodes = (igraph_integer_t) igraph_vcount(graph);
+    igraph_integer_t no_of_edges = (igraph_integer_t) igraph_ecount(graph);
+    igraph_2wheap_t Q;
+    igraph_inclist_t inclist;
+    igraph_adjlist_t fathers;
+    long int source, j;
+    igraph_stack_t S;
+    igraph_neimode_t mode = directed ? IGRAPH_OUT : IGRAPH_ALL;
+    igraph_vector_t dist, nrgeo, tmpscore;
+    igraph_vit_t vit;
+    unsigned char *is_target;
+    int cmp_result;
+    const double eps = IGRAPH_SHORTEST_PATH_EPSILON;
+
+    if (igraph_vector_size(weights) != no_of_edges) {
+        IGRAPH_ERROR("Weight vector length does not match", IGRAPH_EINVAL);
+    }
+    if (no_of_edges > 0) {
+        igraph_real_t minweight = igraph_vector_min(weights);
+        if (minweight <= 0) {
+            IGRAPH_ERROR("Weight vector must be positive", IGRAPH_EINVAL);
+        } else if (igraph_is_nan(minweight)) {
+            IGRAPH_ERROR("Weight vector must not contain NaN values", IGRAPH_EINVAL);
+        } else if (minweight <= eps) {
+            IGRAPH_WARNING("Some weights are smaller than epsilon, calculations may suffer from numerical precision.");
+        }
+    }
+
+    if (!igraph_vs_is_all(&vids)) {
+        /* subset */
+        IGRAPH_CHECK(igraph_vit_create(graph, vids, &vit));
+        IGRAPH_FINALLY(igraph_vit_destroy, &vit); 
+        to_reach = IGRAPH_VIT_SIZE(vit);
+        igraph_vector_resize(res, no_of_nodes);
+        igraph_vector_fill(res, 0);
+    } else {
+        /* only  */
+        return igraph_i_betweenness_cutoff_weighted(graph, res, vids, directed,
+                                                    -1, weights);
+    }
+
+    IGRAPH_CHECK(igraph_2wheap_init(&Q, no_of_nodes));
+    IGRAPH_FINALLY(igraph_2wheap_destroy, &Q);
+    IGRAPH_CHECK(igraph_inclist_init(graph, &inclist, mode, IGRAPH_LOOPS));
+    IGRAPH_FINALLY(igraph_inclist_destroy, &inclist);
+    IGRAPH_CHECK(igraph_adjlist_init_empty(&fathers, no_of_nodes));
+    IGRAPH_FINALLY(igraph_adjlist_destroy, &fathers);
+
+    IGRAPH_CHECK(igraph_stack_init(&S, no_of_nodes));
+    IGRAPH_FINALLY(igraph_stack_destroy, &S);
+    IGRAPH_VECTOR_INIT_FINALLY(&dist, no_of_nodes);
+    IGRAPH_VECTOR_INIT_FINALLY(&tmpscore, no_of_nodes);
+    IGRAPH_VECTOR_INIT_FINALLY(&nrgeo, no_of_nodes);
+
+    is_target = igraph_Calloc(no_of_nodes, unsigned char);
+    if (is_target == 0) {
+        IGRAPH_ERROR("Can't calculate shortest paths", IGRAPH_ENOMEM);
+    }
+    IGRAPH_FINALLY(igraph_free, is_target); 
+
+    for (IGRAPH_VIT_RESET(vit); !IGRAPH_VIT_END(vit); IGRAPH_VIT_NEXT(vit)){
+        if (!is_target[(long int) IGRAPH_VIT_GET(vit)]){
+            is_target[(long int) IGRAPH_VIT_GET(vit)] = 1;
+        }
+        else{
+            to_reach --;
+        }
+    }    
+
+    for (IGRAPH_VIT_RESET(vit); !IGRAPH_VIT_END(vit); IGRAPH_VIT_NEXT(vit)) {
+        source = IGRAPH_VIT_GET(vit);
+        IGRAPH_PROGRESS("Betweenness centrality: ", 100.0 * source / no_of_nodes, 0);
```

Comment:
There is no guarantee that the vertex iterator goes through vertices in order of increasing IDs, therefore this `IGRAPH_PROGRESS` would not increase monotonically.

We should either drop the progress indicator, or use a separate integer variable that is incremented by one in each cycle of the `for` loop.